Add support for AWS Linux and Debian

davidski opened the pull request:
Support the AWS Linux and Debian platforms. Centos not yet supported (requires EPEL).

stevendanna commented on recipes/install_package.rb:
```diff
@@ -19,7 +19,15 @@
+else
+  pkg_name = 'r_base'
+end
+package pkg_name do
```
newline after the `end`

stevendanna commented on recipes/install_package.rb:
```diff
@@ -19,7 +19,15 @@
 #
 
-package 'r-base' do
+case node['platform_family']
```
For things like this where the case statement is choosing between data, I typically go for something like this:

```ruby
pkg_name = case node['platform_family']
           when 'rhel'
             ...
           end
```

stevendanna commented:
Left some minor style comments.  Would you mind rebasing away the "Correct merge conflict" commit?  I can always do that when I merge so no worries if you don't get to it.

Is there not a dev package on Amazon/RHEL systems that we can install?

davidski commented:
Thanks! Will clean this up soon.

There's no dev package that I've found in the Amazon repos, but I haven't checked out EPEL yet.

davidski committed:
Add Linux and AWS Linux support

davidski commented:
Sorry this took so long. All cleaned up and squashed now. Please review and merge if it passes muster. Thanks again!

davidski commented:
Hi, @stevendanna. Will you have time to review and possibly merge this PR soon? I'd like to start working on the testing functions and would rather not have to track multiple PRs. :) Thanks for the continued engagement!

stevendanna commented:
@davidski Heya.  I just pushed a basic travis setup.  I'm running your branch against our basic test-kitchen setup right now with one minor rubocop fix appended.  Once it passes, I'll merge it.

stevendanna commented:
@davidski It looks for the current kitchen tests to pass on centos, we'd have to install EPEL (IIRC this is where the CRAN README's on centos support will send you to).  Since the `yum` cookbook already provides a yum_repository resource, it should be pretty straightforward.

davidski committed:
Merge branch 'master' of https://github.com/stevendanna/cookbook-r into aws_linux

davidski committed:
Move Ubuntu 10->14 and drop centos-6.4 in favor of centos-6.7

davidski committed:
Move Travis CI testing to Ruby 2.2, required by fauxhai and others

davidski committed:
Update rubygems version on Travis to avoid issues with rack gem

davidski committed:
Move to Rvm 2.3 for parity with current ChefDK

davidski committed:
RVM has no 2.3 alias on Travis. Manually specify 2.3.1

davidski committed:
Update rubocop rules to current syntax and perform rubocop and foodcritic linting

davidski committed:
Drop Ubuntu 10. Add Ubuntu 16. Move to bento for boxes

davidski committed:
Ignore .kitchen to allow converge to succeed under windows (long file paths)

davidski committed:
Configure EPEL on centos platforms, as IDed by @stevendana

davidski commented:
Oy. I can't believe I had left this out so long. Trying to close out the year with a few less To Dos. :smile:

I've done a bit of refactoring on this PR.
- Ubuntu 10 dropped and Ubuntu 16 added
- rubocop/foodcritic/cookstyle linting
- Added the EPEL repository via the supermarket yum-epel recipe
- Updated Travis build environment for some ruby dependencies

Hopefully you'll be able to review and merge a bit faster than I was in getting this back to you. 😁

sfromm-plm commented:
For centos-7.2, I had to change `default['r']['install_dir']` in `attributes/install.rb` to `'/usr/lib64/R'`.

stevendanna commented:
@davidski Thanks for pulling this together.  I'll take a look and try to get another release out this month.

davidski commented:
Hey there, @stevendanna. Might you be able to review and possibly merge this soon? Was dealing with a StackOverflow question recently where I think this would have been helpful. Thanks a bunch!

stevendanna commented:
@davidski Yup, sorry I forgot about this.  I put it on my todo list for the week.

stevendanna commented:
I went ahead and merged this as is, I'll try to take a look at this cookbook tomorrow and get it back in shape.

davidski commented:
Sorry for missing this thread, @stevendanna! The Travis builds look to be passing (https://travis-ci.org/stevendanna/cookbook-r/builds/207778940 is green). Is there anything I'm missing here that I can help out with?
